packages/teams-js/test/public/media.spec.ts

```diff
@@ -508,67 +508,56 @@ describe('media', () => {
+        await expect(promise).rejects.toEqual({ errorCode: ErrorCode.OLD_PLATFORM });
       });
 
-      it('videoController notifyEventToHost is not handled successfully', () => {
-        return mobilePlatformMock.initializeWithContext(FrameContexts.content, HostClientType.android).then(() => {
-          mobilePlatformMock.setClientSupportedSDKVersion(nonFullScreenVideoModeAPISupportVersion);
-          let mediaError: SdkError;
-          new media.VideoController().stop((e: SdkError) => {
-            mediaError = e;
-          });
-
-          const err: SdkError = {
-            errorCode: ErrorCode.INTERNAL_ERROR,
-          };
-          const message = mobilePlatformMock.findMessageByFunc('media.controller');
-          expect(message).not.toBeNull();
-          expect(message.args.length).toBe(1);
+      it('videoController notifyEventToHost is handled successfully', async () => {
+        await mobilePlatformMock.initializeWithContext(FrameContexts.content, HostClientType.android);
+        mobilePlatformMock.setClientSupportedSDKVersion(nonFullScreenVideoModeAPISupportVersion);
+        let mediaError: boolean;
+        new media.VideoController().stop(err => (mediaError = err));
+        const message = mobilePlatformMock.findMessageByFunc('media.controller');
+        expect(message).not.toBeNull();
+        expect(message.args.length).toBe(1);
+        const callbackId = message.id;
+        mobilePlatformMock.respondToMessage({
+          data: {
+            id: callbackId,
+            args: [undefined],
+          },
+        } as DOMMessageEvent);
+        expect(mediaError).toBeFalsy();
+      });
 
-          const callbackId = message.id;
-          mobilePlatformMock.respondToMessage({
-            data: {
-              id: callbackId,
-              args: [err],
-            },
-          } as DOMMessageEvent);
+      it('videoController notifyEventToHost is not handled successfully', async () => {
+        await mobilePlatformMock.initializeWithContext(FrameContexts.content, HostClientType.android);
+        mobilePlatformMock.setClientSupportedSDKVersion(nonFullScreenVideoModeAPISupportVersion);
 
-          expect(mediaError).toBe(err);
+        let mediaError: boolean;
+        new media.VideoController().stop((e: boolean) => {
+          mediaError = e;
         });
+        const err = jest.fn().mockImplementation(() =>
+          Promise.reject({
+            errorCode: ErrorCode.INTERNAL_ERROR,
+          }),
+        );
+        const message = mobilePlatformMock.findMessageByFunc('media.controller');
+        expect(message).not.toBeNull();
+        expect(message.args.length).toBe(1);
+        const callbackId = message.id;
+        mobilePlatformMock.respondToMessage({
+          data: {
+            id: callbackId,
+            args: [err],
+          },
+        } as DOMMessageEvent);
+        await expect(mediaError).rejects.toEqual({ errorCode: ErrorCode.INTERNAL_ERROR });
```

Comment:
>rejects [](http://example.com/codeflow?start=33&length=7)

What does it mean to use rejects when evaluating a boolean? I thought rejects was for promises

Reply:
Actually I think this whole line doesn't work because you're expecting a boolean to equal an SdkError